### PR TITLE
Use local binder logo svg

### LINKED ISSUE #735: Binder icon is hardcoded, which causes a loading failure with on some browsers
See https://github.com/scikit-image/scikit-image/issues/4748

It would be helpful if this line could be changed: https://github.com/sphinx-gallery/sphinx-gallery/blob/master/sphinx_gallery/binder.py#L109
Because the icon is hosted on a third-party website, browsers with "prevent cross-site tracking" preferences remove the icon.

Thank you!

### PR DESCRIPTION
**Pull request by lucyleeow**
closes #735 

Binder logo is copied from packaged `_static` folder to `target_dir`/images folder (not sure if this is best place?).
Added test.

### CODE REVIEW

File: sphinx_gallery/binder.py [hunk 1]
```diff
@@ -103,12 +111,26 @@ def gen_binder_rst(fpath, binder_conf, gallery_conf):
...
+            shutil.copyfile(binder_logo, binder_logo_full_path)
+    sources_dir = gallery_conf['src_dir']
+    binder_logo_rel_path = os.path.relpath(binder_logo_full_path, sources_dir)
+    binder_logo_rel_path = binder_logo_rel_path.replace(os.sep, '/')
+
     rst = (
         "\n"
         "  .. container:: binder-badge\n\n"
-        "    .. image:: https://mybinder.org/badge_logo.svg\n"
+        "    .. image:: /{}\n"
```
Comment (lucyleeow) [inline sphinx_gallery/binder.py:131]: I copied this from:
https://github.com/sphinx-gallery/sphinx-gallery/blob/b09770c9dae3eeb8c10b5e45b70fd19f9db70cd1/sphinx_gallery/scrapers.py#L429

but I don't understand how the `/` works for windows OS?

Comment (larsoner) [reply sphinx_gallery/binder.py:131]: I think it should be fine. IIRC sphinx makes it relative to the build root, so I'd expect something like `/_static/binder_badge_logo.svg` could be used directly

File: sphinx_gallery/binder.py [hunk 2]
```diff
@@ -103,12 +111,26 @@ def gen_binder_rst(fpath, binder_conf, gallery_conf):
...
+    binder_logo = os.path.join(glr_path_static(), 'binder_badge_logo.svg')
+    binder_logo_full_path = os.path.join(
+        target_dir, 'images', 'binder_badge_logo.svg'
+    )
+    if gallery_conf['gallery_dirs'] is None:
+        # Avoid copying file when testing
+        pass
+    else:
+        if not os.path.exists(binder_logo_full_path):
+            shutil.copyfile(binder_logo, binder_logo_full_path)
```
Comment (larsoner) [inline sphinx_gallery/binder.py:123]: and if you use `_static`, no need to `copyfile`

Comment (lucyleeow) [reply sphinx_gallery/binder.py:123]: Can you clarify? As in I can use the logo in the packaged static file e.g.
```
.. image:: /home/lucy/Documents/sphinx-gallery/sphinx_gallery/_static/binder_badge_logo.svg
``` 
?

Comment (lucyleeow) [reply sphinx_gallery/binder.py:123]: You're right, that would have saved a lot of time!

Comment (larsoner) [reply sphinx_gallery/binder.py:123]: I would try putting just `/_static/binder_badge_logo.svg` here. If it doesn't work, then doing `os.path.join(os.path.relpath(src_dir, this_file_dir), '_static', 'binder_badge_logo.svg')` or so should work. Basically it already lives in `_static` so we should just reuse it

Comment (lucyleeow) [reply sphinx_gallery/binder.py:123]: > /_static/binder_badge_logo.svg

that doesn't work but the full path works

Comment (larsoner) [reply sphinx_gallery/binder.py:123]: Sure, whatever path works is fine by me!

**Comment by larsoner** [comment]
Lots of these on CircleCI:
```
/home/circleci/project/doc/auto_examples/local_module.rst:: WARNING: image file not readable: home/circleci/project/sphinx_gallery/_static/binder_badge_logo.svg
/home/circleci/project/doc/auto_examples/no_output/just_code.rst:: WARNING: image file not readable: home/circleci/project/sphinx_gallery/_static/binder_badge_logo.svg
/home/circleci/project/doc/auto_examples/no_output/plot_raise.rst:: WARNING: image file not readable: home/circleci/project/sphinx_gallery/_static/binder_badge_logo.svg
```
might be worth adding a `assert op.isfile(...)` to test_full to ensure it's copied properly

**Comment by lucyleeow** [comment]
Actually I was mistaken. Using the packaged file doesn't work locally. I guess sphinx issue?

**Comment by lucyleeow** [comment]
I get the same error locally:
```
/home/lucy/Documents/sphinx-gallery/doc/auto_examples/local_module.rst:: WARNING: image file not readable: home/lucy/Documents/sphinx-gallery/sphinx_gallery/_static/binder_badge_logo.svg
```

**Comment by larsoner** [comment]
Do you want me to try and push if I figure out the correct path?

**Comment by lucyleeow** [comment]
I think I just got it!

**Comment by larsoner** [comment]
Nice, some `relpath` magic :)

**Comment by lucyleeow** [comment]
Thanks @larsoner !

**Pull request merged**